pkg/snapshot/restorer/restorer.go

```diff
@@ -126,7 +126,10 @@ func (r *Restorer) Restore(ro brtypes.RestoreOptions, m member.Control) (*embed.
 				r.logger.Errorf("failed to close etcd cluster client: %v", err)
 			}
 		}()
-		m.UpdateMemberPeerURL(context.TODO(), clientCluster)
+
+		if err := m.UpdateMemberPeerURL(context.TODO(), clientCluster); err != nil {
+			return e, err
+		}
```

Comment:
Is it make sense to fail the restoration if backup-restore fail to update the `PeerURL` after restoration ? what if we just log the warning ?

Reply:
Failing to update peer URLs is definitely an error, rather than a warning. If clustering fails, there's no use starting the etcd member in the first place. And it would be difficult to debug such an error, because the etcd config file will look fine but runtime config wasn't updated, and going through warning logs isn't ideal. I would recommend throwing an error and failing early in such cases.

Reply:
> Failing to update peer URLs is definitely an error, rather than a warning

yeah, correct then we can just log an error rather than a warning.

> If clustering fails, there's no use starting the etcd member in the first place. And it would be difficult to debug such an error, because the etcd config file will look fine but runtime config wasn't updated

this update to member Peer URL is only introduce for multi-node (recovering from permanent quorum loss),
so for `cluster size =1`, it won't cause an issue even if this call get failed.
for `HA cluster`, it can cause an issue while doing recovering from permanent quorum loss, so I don't have any hard opinion on this.  If you want to return the `err` then it's fine to me, just test the scenario for restoration for `cluster size =1`. Hope that's fine.

Reply:
Sure. I would like to retain the returning of the error.